### ISSUE TITLE
fatal error: api/peerconnectioninterface.h: No such file or directory

### ISSUE THREAD

**Issue by dsnopek** [godotengine/webrtc-native#4]
I've been trying to compile this with the latest Godot sources, and I've been getting this error:

```
In file included from src/WebRTCLibDataChannel.cpp:1:0:
src/WebRTCLibDataChannel.hpp:6:10: fatal error: api/peerconnectioninterface.h: No such file or directory
 #include "api/peerconnectioninterface.h" // interface for all things needed from WebRTC
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I can't seem to find the peerconnectioninterface.h file in the Godot source, or godot-cpp or godot_headers. What am I missing?

**Comment by dsnopek** [comment]
Ah, is this the library that needs to get put in `lib/webrtc`:

https://github.com/aisouard/libwebrtc

Still building it, but I'll see if I can get the resulting libraries in the right place and get the GDNative library building.

**Comment by dsnopek** [comment]
Hm. Trying to compile https://github.com/aisouard/libwebrtc leads to this error:

```
0> Failed to fetch file gs://chromium-gn/2d79b4acffac0b084f250a9699f670d41f164864 for src/buildtools/linux64/gn, skipping. [Err: /home/dsnopek/prj/libwebrtc/out/depot_tools/src/depot-tools/external_bin/gsutil/gsutil_4.26/gsutil/third_party/boto/boto/pyami/config.py:69: UserWarning: Unable to load AWS_CREDENTIAL_FILE ()
  warnings.warn('Unable to load AWS_CREDENTIAL_FILE (%s)' % full_path)
Failure: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727).
]
```

I Google'd a bunch of other people having that error some years ago, but I couldn't find a solution yet...

**Comment by dsnopek** [comment]
Or should I be working off of this one?

https://webrtc.org/native-code/development/

Seems more "upstream-y" but it's harder to get started, more tools and things are needed. I'll dig into it more when I have chance...

**Comment by Faless** [comment]
You need to use branch 68
https://chromium.googlesource.com/external/webrtc/+/branch-heads/68 .
I suggest you use this script for building:
https://github.com/sourcey/webrtc-builds
Or even use the pre-compiled libwebrtc in this branch release:
https://github.com/Faless/webrtc-builds/releases/tag/23360-350e48a

**Comment by dsnopek** [comment]
Thanks! I'll give that a try later

**Comment by dsnopek** [comment]
Using https://github.com/sourcey/webrtc-builds seemed to work! I copied the resulting files into lib/webrtc, and I got past that original error!

I'm still getting some compilation error, but it's back in Godot stuff, and I'm still looking into it. In any case, it's not related to this issue anymore! If I can't work out this new error, I'll open a new issue. :-)

**Comment by shintiger** [comment]
How do you build https://github.com/sourcey/webrtc-builds with branch-head/68?
I got error:Cound not get branch revision

**Comment by token-cjg** [comment]
I encountered the same problem as shintiger.  After managing to get things to build, I came across the error in this issue, then searched the issues for this project and came across this thread.

Some amateur sleuthing for the above error suggests that it looks like the old way of getting Chromium code [has been obsoleted](https://sites.google.com/a/chromium.org/dev/developers/how-tos/old-get-the-code).  This uses things that look familiar from this codebase, like ninga, gclient, out directories, etc.

Therefore it appears that scripts will need to be updated to get the Chromium code [using this new approach](https://chromium.googlesource.com/chromium/src/+/master/docs/mac_build_instructions.md).

I'll file a separate issue for this in this repo, as this one appears to have been closed.
